src/validators/tuple.rs

```diff
@@ -0,0 +1,127 @@
+use pyo3::prelude::*;
+use pyo3::types::{PyDict, PyTuple};
+
+use crate::build_tools::{is_strict, SchemaDict};
+use crate::errors::{context, err_val_error, ErrorKind, InputValue, LocItem, ValError, ValLineError};
+use crate::input::{GenericSequence, Input, SequenceLenIter};
+
+use super::{build_validator, BuildContext, BuildValidator, CombinedValidator, Extra, ValResult, Validator};
+
+#[derive(Debug, Clone)]
+pub struct TupleValidator {
```

Comment:
We need two separate types of `TupleSchema` (same as many other types like `int` have multiple schemas):

* for `tuple[str, ...]` - should be like this
* for `tuple[str, bool, int]` - should have `Vec<CombinedValidator>`

Reply:
@samuelcolvin 
Understood, thank you.
Should the tuple with individual elements' types specified have validation relaxed when non-strict?

Reply:
No.